library/src/scripts/components/siteNav/SiteNav.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import { t } from "../../application";
+import classNames from "classnames";
+import { getRequiredID } from "@library/componentIDs";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import SiteNavNode from "@library/components/siteNav/SiteNavNode";
+import TabHandler from "@library/TabHandler";
+
+interface IProps extends RouteComponentProps<{}> {
+    name: string;
+    className?: string;
+    children: any[];
+}
+
+export interface IState {
+    id: string;
+}
+
+/**
+ * Recursive component to generate site nav
+ * No need to set "counter". It will be set automatically. Kept optional to not need to call it on the top level. Used for React's "key" values
+ */
+export class SiteNav extends React.Component<IProps, IState> {
+    public constructor(props) {
+        super(props);
+        this.state = {
+            id: getRequiredID(props, "siteNav"),
+        };
+    }
+
+    public get titleID() {
+        return this.state.id + "-title";
+    }
+
+    public render() {
+        const content =
+            this.props.children && this.props.children.length > 0
+                ? this.props.children.map((child, i) => {
+                      return (
+                          <SiteNavNode
+                              {...child}
+                              key={`navNode-${i}`}
+                              counter={i}
+                              titleID={this.titleID}
+                              visible={true}
+                              location={this.props.location}
+                              depth={0}
+                          />
+                      );
+                  })
+                : null;
+        return (
+            <nav onKeyDownCapture={this.handleKeyDown as any} className={classNames("siteNav", this.props.className)}>
```

Comment:
Can we remove this `any` assertion? Changing the type of the event in our handler to `React.KeyboardEvent` will fix the type-checking.